pkg/util/trace/trace.go

```diff
@@ -0,0 +1,49 @@
+package trace
+
+import (
+	"sync"
+	"time"
+
+	"k8s.io/utils/trace"
+)
+
+type Tracer struct {
+	traceMap map[string]*trace.Trace
+	mux      sync.Mutex
+
+	Threshold time.Duration
+}
+
+func (t *Tracer) StartTrace(key string, name string, field ...trace.Field) {
+	t.mux.Lock()
+	defer t.mux.Unlock()
+	if t.traceMap == nil {
+		t.traceMap = make(map[string]*trace.Trace)
+	}
+	t.traceMap[key] = trace.New(name, field...)
+	return
+}
+
+func (t *Tracer) StopTrace(key string) {
+	t.mux.Lock()
+	defer t.mux.Unlock()
+	t.traceMap[key].LogIfLong(t.Threshold)
```

Comment:
i think it would be worthwhile to ensure the key exists in the map here just in case this library is misused. Otherwise, this line would crash the process if the key doesn't exist anymore

Reply:
Sure.